apps/nextra/pages/en/build/create-aptos-dapp/templates/custom-indexer.mdx

```diff
@@ -0,0 +1,219 @@
+
+### Setup the database
+
+Everything here should be run in the `indexer` folder.
+
+Install diesel cli to run migrations.
+
+```sh
+cargo install diesel_cli --no-default-features --features postgres
+```
+
+Run all pending database migrations. This will create all the tables in the database.
+
+```sh
+diesel migration run \
+    --config-file="src/db_migrations/diesel.toml" \
+    --database-url="postgresql://username:password@neon_host/db_name?sslmode=require"
+```
+
+In case you want to revert all migrations to delete all tables because you want to re-index all data.
+
+```sh
+diesel migration revert \
+	--all \
+	--config-file="src/db_migrations/diesel.toml" \
+    --database-url="postgresql://username:password@neon_host/db_name?sslmode=require"
+```
+
+In case you want to make any change to the database, you can add a new migration.
+
+```sh
+diesel migration generate create-abc-table \
+    --config-file="src/db_migrations/diesel.toml"
+```
+
+### Sign up for Aptos Build
+
+Sign up for [Aptos Build](https://developers.aptoslabs.com/), create a new project and get the API token.
+
+### Run the custom indexer locally
+
+Make a copy of `example.config.yaml` of `indexer` folder and rename it to `config.yaml`. Follow the comments to fill in the 
+- `starting_version`: The tx version (Aptos concept, similar to block height) from which you want to start indexing
+- `postgres_connection_string`: The connection string of the postgres database
+- `contract_address`: The address of the Move contract
+- `auth_token`: Aptos Build API token
+
+Run the below command to run the custom indexer locally:
+
+```sh
+cargo run --release -- -c config.yaml
+```
+
+### Run the frontend
+
+```bash
+npm run dev
+```
+
+</Steps>
+
+### Building the frontend
+
+The boilerplate template utilizes React as the frontend framework and Next.js as the development tool, and is styled with [Tailwind CSS](https://tailwindcss.com/) and [shadcn/ui](https://ui.shadcn.com/).
+All dapp components should be added into the `components` folder and it is recommended to create a `app` folder to hold all future pages in your project.
+
+### Writing a Move contract
+
+The boilerplate template comes with a `contract` folder that holds all Move smart contract related files. Under the `sources` folder you will find a `*.move` file with a super basic implementation of a Move module that stores a message and updates it. This is to help you get started with writing your own Smart Contract.
+
+### Smart contract and frontend communication
+
+For a frontend to submit a transaction to a smart contract, it needs to call an entry function. The boilerplate provides you with an `entry-functions` folder to hold all your dapp entry function requests.
+Additionaly, for a frontend to fetch data from a smart contract, it reads data from the Postgres database.
```

Comment:
where is the logic to read the data from Postrgres?

Reply:
There's a db folder in frontend that has all db queries

Reply:
lets mention it then